### PR TITLE
Update ZBOSS libraries to release v3.9.0.0 && remove Zigbee GP Combo Basic libraries

### PR DESCRIPTION
**Pull request by sebastiandraus**

### CODE REVIEW

File: zboss/CHANGELOG.rst [hunk 1]
```diff
@@ -9,6 +9,22 @@ Changelog
...
+
+Changes
+=======
+
+* Added APS commands dups detection.
+* Fixed ZR going out of capacity due to many joins/rejoins.
+* Added NCP protocol version checks.
+* Fixed unsolicited NCP commands not being handled correctly.
+* Removed precompiled libraries for Zigbee Green Power Combo Basic functionality.
+* Updated the ZBOSS stack to version ``3.9.0.0+4.0.1``.
```
Comment (tomchy) [inline zboss/CHANGELOG.rst:25]: `3.9.0.1` due to documentation build issues?

File: zboss/include/osif/zb_revision_r22.h [hunk 2]
```diff
@@ -29,7 +29,7 @@
  *
  * Note: This value is automatically updated by the release script.
  */
-#define ZBOSS_PLATFORM_PATCH 0U
+#define ZBOSS_PLATFORM_PATCH 1U
```
Comment (tomchy) [inline zboss/include/osif/zb_revision_r22.h:32]: What is the reason for having the PATCH value set to 1?

File: zboss/CHANGELOG.rst (hunk 1 above)
Comment (sebastiandraus) [reply zboss/CHANGELOG.rst:25]: Will be changed - waiting for new sdk to be built

File: zboss/include/osif/zb_revision_r22.h (hunk 2 above)
Comment (sebastiandraus) [reply zboss/include/osif/zb_revision_r22.h:32]: Not sure, will check this

File: zboss/CHANGELOG.rst (hunk 1 above)
Comment (sebastiandraus) [reply zboss/CHANGELOG.rst]: Changed to 3.9.0.1, marking as resolved

File: zboss/include/osif/zb_revision_r22.h (hunk 2 above)
Comment (sebastiandraus) [reply zboss/include/osif/zb_revision_r22.h]: Patch value kept at 0, platform minor version set to 1U due to changes with NCP prot version

File: zboss/doc/zboss_configuration.rst [hunk 3]
```diff
@@ -33,4 +33,5 @@ To enable additional features in the ZBOSS libraries, you can use the following
   This option is enabled by default in the :ref:`Zigbee NCP sample <nrf:zigbee_ncp_sample>`.
   This option uses a production version of ZBOSS that has not been certified.
 * :kconfig:`CONFIG_ZIGBEE_GP_CB` - With this option enabled, the application can support the Green Power Combo feature, which implements the basic set of Green Power Proxy and Green Power Sink functionalities within a single device.
+  This option requires access to ZBOSS stack sources to be enabled.
```
Comment (greg-fer) [inline zboss/doc/zboss_configuration.rst:36]: What does this mean, in practice?

File: zboss/CHANGELOG.rst [hunk 4]
```diff
@@ -9,6 +9,22 @@ Changelog
...
+Changes
+=======
+
+* Added APS commands dups detection.
+* Fixed ZR going out of capacity due to many joins/rejoins.
+* Added NCP protocol version checks.
+* Fixed unsolicited NCP commands not being handled correctly.
+* Removed precompiled libraries for Zigbee Green Power Combo Basic functionality.
+* Updated the ZBOSS stack to version ``3.9.0.1+4.1.0``.
+  For detailed information, see `ZBOSS stack release notes`_ for the main branch.
```
Comment (greg-fer) [inline zboss/CHANGELOG.rst:26]:
```suggestion
Added
=======

* Added APS command duplicate detection.
* Added NCP protocol version checks.

Updated
=======

* Updated the ZBOSS stack to version ``3.9.0.1+4.1.0``.
  For detailed information, see `ZBOSS stack release notes`_ for the main branch.
* Removed precompiled libraries for Zigbee Green Power Combo Basic functionality.
  
Bug fixes
=========

 * Fixed an issue where Zigbee Router would go out of capacity due to too many joins or rejoins.
 * Fixed an issue where unsolicited NCP commands would not be handled correctly.
```

**Review by greg-fer — CHANGES_REQUESTED** [review]

File: zboss/doc/zboss_configuration.rst (hunk 3 above)
Comment (sebastiandraus) [reply zboss/doc/zboss_configuration.rst:36]: Green Power Combo Basic functionality can be used only if zigbee samples are built with ZBOSS stack from sources

File: zboss/CHANGELOG.rst (hunk 4 above)
Comment (sebastiandraus) [reply zboss/CHANGELOG.rst]: Done

File: zboss/doc/zboss_configuration.rst (hunk 3 above)
Comment (greg-fer) [reply zboss/doc/zboss_configuration.rst:36]: OK, can you phrase this sentence in this manner? Thanks.

**Review by greg-fer — APPROVED** [review]
Approving, as I'll be out until Monday, but please integrate my last remaining comment.

**Review by tomchy — APPROVED** [review]

File: zboss/doc/zboss_configuration.rst (hunk 3 above)
Comment (sebastiandraus) [reply zboss/doc/zboss_configuration.rst]: done!

**Comment by sebastiandraus** [comment]
Rebased at newest main

**Review by rlubos — APPROVED** [review]

**Pull request merged**